UnityGLTF/Assets/UnityGLTF/Scripts/MemoryChecker.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using UnityEngine;
+#if WINDOWS_UWP
+using Windows.System;
+#endif
+
+namespace UnityGLTF
+{
+	public class MemoryChecker
+	{
+		private bool outOfMemory = false;
+
+		/// <summary>
+		/// Allows polling for app low memory situation.  Listens to Application.lowMemory, which works for iOS and Android.
+		/// Also listens to Windows-specific MemoryManager.AppMemoryUsageIncreased.
+		/// </summary>
+		public MemoryChecker()
+		{
+			Application.lowMemory += Application_lowMemory;
+
+#if WINDOWS_UWP
+			MemoryManager.AppMemoryUsageIncreased += MemoryManager_AppMemoryUsageIncreased;
+#endif
+		}
+
+		/// <summary>
+		/// If the OS has notified the app that it is low on memory since MemoryChecker was constructed, this method will throw an OutOfMemoryException.
+		/// Once it throws once, it will continue to throw.  Callers are advised to construct a new MemoryChecker after memory usage has gone down.
+		/// </summary>
+		public void ThrowIfOutOfMemory()
+		{
+			if (outOfMemory)
+			{
+				throw new OutOfMemoryException();
+			}
+		}
+
+		private void Application_lowMemory()
+		{
+			outOfMemory = true;
+		}
+
+#if WINDOWS_UWP
+		private void MemoryManager_AppMemoryUsageIncreased(object sender, object e)
+		{
+			if (MemoryManager.AppMemoryUsageLevel == AppMemoryUsageLevel.OverLimit)
+			{
+				outOfMemory = true;
```

Comment:
Is there a way to check memory again and set this to false if memory gets cleaned up?

Reply:
Unfortunately, no.  It is not really possible.  The underlying events from Android and Windows _do_ tell you when they are no longer low on memory.  But it doesn't appear that iOS exposes an event like that.  So Unity does not expose any event for it.  The expectation is that the app must reduce its memory usage when that event fires.  If your process is not terminated, then I guess that means that you're no longer low on memory?

I considered setting outOfMemory to false when throwing, assuming that it will go down and then could go up again and then we'd get the event again, but I don't know if that would work well.

So I decided to put the comment at the top explaining the expected usage.  If it throws, then the caller should clear some memory and create a new MemoryChecker.